cli/cmd/upgrade.go

```diff
@@ -507,6 +507,50 @@ func fetchTLSSecret(k *k8s.KubernetesAPI, webhook string, options *upgradeOption
 	return value, nil
 }
 
+func fetchK8sTLSSecret(k *k8s.KubernetesAPI, webhook string, options *upgradeOptions) (*charts.TLS, error) {
+	webhookSecretNameStr := webhookK8sSecretName(webhook)
+
+	secret, err := k.CoreV1().
+		Secrets(controlPlaneNamespace).
+		Get(webhookSecretNameStr, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	if secret.Data == nil || secret.Data["tls.key"] == nil || secret.Data["tls.crt"] == nil {
+		return nil, fmt.Errorf("TLS Secret %s is missing data fields (data.tls.key and data.tls.crt expected)", webhookSecretNameStr)
+	}
+
+	value := &charts.TLS{
+		KeyPEM: string(secret.Data["tls.key"]),
+		CrtPEM: string(secret.Data["tls.crt"]),
+	}
+
+	if err := injectCABundle(k, webhook, value); err != nil {
+		return nil, err
+	}
+
+	if err := options.verifyTLS(value, webhook); err != nil {
+		return nil, err
+	}
+
+	return value, nil
+}
+
+// try to fetch K8sTLSSecret, but revert to TLSSecret if the former fails
+func fetchOrUpdateTLSSecret(k *k8s.KubernetesAPI, webhook string, options *upgradeOptions) (*charts.TLS, error) {
```

Comment:
It looks like this just fetches the TLS data from the secret in one format or another.  Let's rename this to not use the word "Update" since this doesn't update anything.

`fetchTLSSecertWithFallback`? Or call this `fetchTLSSecret` and rename the existing `fetchTLSSecret` to `fetchLinkerdTLSSecret`?